platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalPreferenceDaoImpl.java

```diff
@@ -43,92 +35,40 @@
-                    Map<String, Map<String, String>> m = gson.fromJson(reader, new TypeToken<Map<String, Map<String, String>>>() {
-                    }.getType());
-                    if (m != null) {
-                        storage.putAll(m);
-                    }
-                } catch (Exception e) {
-                    LOG.error(String.format("Failed load user profiles form %s", storageFile), e);
-                } finally {
-                    if (reader != null) {
-                        try {
-                            reader.close();
-                        } catch (IOException ignored) {
-                        }
-                    }
-                }
-            }
-            // Add default entry if file doesn't exist or invalid or empty.
-            if (storage.isEmpty()) {
-                final Map<String, String> newPreferences = new HashMap<>(4);
-                newPreferences.put("preference1", "value");
-                newPreferences.put("preference2", "value");
-                storage.put("codenvy", newPreferences);
-            }
-        } finally {
-            lock.writeLock().unlock();
+        preferences.putAll(preferenceStorage.loadMap(new TypeToken<Map<String, Map<String, String>>>() {}));
+        // Add default entry if file doesn't exist or invalid or empty.
+        if (preferences.isEmpty()) {
+            final Map<String, String> newPreferences = new HashMap<>(4);
+            newPreferences.put("preference1", "value");
+            newPreferences.put("preference2", "value");
+            preferences.put("codenvy", newPreferences);
         }
     }
 
     @PreDestroy
-    private void stop() {
-        lock.writeLock().lock();
-        try {
-            Writer writer = null;
-            try {
-                writer = Files.newWriter(storageFile, Charset.forName("UTF-8"));
-                gson.toJson(storage, new TypeToken<Map<String, Map<String, String>>>() {
-                }.getType(), writer);
-            } catch (Exception e) {
-                LOG.error(String.format("Failed setPreferences user profiles form %s", storageFile), e);
-            } finally {
-                if (writer != null) {
-                    try {
-                        writer.close();
-                    } catch (IOException e) {
-                        LOG.error(String.format("Failed setPreferences user profiles form %s", storageFile), e);
-                    }
-                }
-            }
-        } finally {
-            lock.writeLock().unlock();
-        }
+    private void stop() throws IOException {
+        preferenceStorage.store(preferences);
     }
 
     @Override
     public void setPreferences(String userId, Map<String, String> preferences) throws ServerException, NotFoundException {
         lock.writeLock().lock();
         try {
-            storage.put(userId, new HashMap<>(preferences));
+            this.preferences.put(userId, new HashMap<>(preferences));
```

Comment:
it is too easy to make mistake here. What if rename one of the preferences variables ?